sphinx/util/console.py

```diff
@@ -75,9 +81,24 @@ def nocolor() -> None:
 
 
 def coloron() -> None:
+    if sys.platform == 'win32' and colorama is not None:
+        colorama.init()
     codes.update(_orig_codes)
 
 
+@contextmanager
+def colorize_output(on: bool) -> Generator[None, None, None]:
+    """Temporarily enable or disable colorized output streams."""
+    global codes
+    current_codes = codes.copy()
+    if on:
+        coloron()
+    else:
+        nocolor()
+    yield
```

Comment:
Based on the docstring: should there be toggle actions both before and after the `yield` statement?

```suggestion
    (coloron if on else nocolor)()
    yield
    (nocolor if on else coloron)()
```

...or something neater (example for illustration purposes, not necessarily good code).

Reply:
maybe toggle is the wrong word then; it should just turn on/off colorization, at the start of the context, then return it to the initial state on exiting the context

Reply:
I'm a little wary of thread-safety / potential timing issues related to the `codes` object, but thanks for resolving my concern about the content of this function.  The code as you have it is OK with me :+1:

Reply:
Oh I'm not a fan of any mutable global variables, I wish there wasn't so many in docutils/sphinx 😅